compiler/src/dotty/tools/dotc/transform/UnusedDecls.scala

```diff
@@ -0,0 +1,49 @@
+package dotty.tools.dotc.transform
```

Comment:
It seems to be quote a bit of code. Would the same effect not be obtained more cheaply by rolling this into erasure?

Reply:
The idea was to prune the tree early to reduce the cost of traversing those declarations. I would have to check if it can be placed in erasure.